service/src/integrationTest/kotlin/fi/espoo/evaka/pedagogicaldocument/PedagogicalDocumentIntegrationTest.kt

```diff
@@ -112,13 +119,13 @@ class PedagogicalDocumentIntegrationTest : FullApplicationTest(resetDbBeforeEach
         }
     }
 
-    private fun createDocumentAsUser(childId: ChildId, user: AuthenticatedUser) =
-        http
-            .post("/employee/pedagogical-document")
-            .jsonBody("""{"childId": "$childId", "description": ""}""")
-            .asUser(user)
-            .responseString()
-            .third
+    private fun createDocumentAsUser(childId: ChildId, user: AuthenticatedUser.Employee) =
+        pedagogicalDocumentController.createPedagogicalDocument(
+            dbInstance(),
+            user,
+            clock,
+            PedagogicalDocumentPostBody(childId, ""),
+        )
```

Comment:
good refactoring of legacy code 👍 could consider updating the other similar things in this file too if you want to